### ISSUE TITLE
Show individual rolls with Better Rolls gives error

### ISSUE THREAD

**Issue by Zamrod** [Stendarpaval/mob-attack-tool#10]
When I have individual rolls turned on and display individual rolls, It does not display anything but gives the following error:

2fields.js:159 Uncaught (in promise) TypeError: Cannot read property 'name' of null
    at Function.constructAttackRoll (fields.js:159)
    at Function.constructModelsFromField (fields.js:436)
    at CustomItemRoll._processField (custom-roll.js:922)
    at CustomItemRoll.roll (custom-roll.js:664)
    at async CustomItemRoll.toMessage (custom-roll.js:770)
    at async rollMobAttackIndividually (mobAttackTool.js:332)

**Comment by Stendarpaval** [comment]
Thanks for sharing the error message. I've not been able to replicate this error, however. It'd really help me if you could answer these questions:

- What kind of weapon or weapons did you use? 
- From the error message it seems that you use the Better Rolls for 5e module. Do you also use midi-QOL? 
- Does everything work as you'd expect if you temporarily disable Better Rolls for 5e?

I'll try to think of a way to prevent such errors from happening in the meantime.

**Comment by Stendarpaval** [comment]
Since your error points towards part of Better Rolls's `field.js` file relating to ammo in custom attack rolls, I've made a small patch to properly check if the weapon uses ammo and released this patch in [v0.1.14](https://github.com/Stendarpaval/mob-attack-tool/releases/tag/v0.1.14). 

So in addition to answering the questions above, could you update Mob Attack Tool and let me know if the issue persists? Thank you.

**Comment by Zamrod** [comment]
> * What kind of weapon or weapons did you use?
6 goblins using their short bow. When testing it using their scimitars it works better, but it shows the attack roll ad 0d0 + 17 when two attacks hit. It did show both the damage rolls.
> * From the error message it seems that you use the Better Rolls for 5e module. Do you also use midi-QOL?
Both of them were enabled when I ran the test, however I get the same error when disabling all modules except mob attack (0.1.13) and better rolls (1.4.5).
> * Does everything work as you'd expect if you temporarily disable Better Rolls for 5e?
If I disable Better Rolls, it works but it doesn’t display the Individual attack rolls (which the settings sound like you need to have it enabled to use).

It looks like the custom roll being sent just doesn’t have a name label being sent with it so it is failing.

**Comment by Zamrod** [comment]
Same error with the new version

**Comment by Stendarpaval** [comment]
> If I disable Better Rolls, it works but it doesn’t display the Individual attack rolls (which the settings sound like you need to have it enabled to use).

Good point, I need to update the settings text because individual attack rolls now also show without Better Rolls (though not when Midi-QOL is active without Better Rolls). 

> It looks like the custom roll being sent just doesn’t have a name label being sent with it so it is failing.

> Same error with the new version

Alright, I'll try to add a name to the ammo, then. Just a moment...

**Comment by Stendarpaval** [comment]
Alright, could you give this patched version a try @Zamrod ? [v0.1.15](https://github.com/Stendarpaval/mob-attack-tool/releases/tag/v0.1.15)

**Comment by Zamrod** [comment]
I get this error with only BR and mob in the new version:
mobAttackTool.js:332 Uncaught (in promise) TypeError: Cannot read property 'name' of null
    at rollMobAttackIndividually (mobAttackTool.js:332)

**Comment by Stendarpaval** [comment]
There seems to be an issue accessing the `name` attribute of the ammunition that you're using for these goblin shortbows. Are the goblins' shortbows set up to use ammo of some kind, which they have run out of somehow?

I've also made yet another patch, [v0.1.16](https://github.com/Stendarpaval/mob-attack-tool/releases/tag/v0.1.16), in which I've added some error handling to the sections that deal with ammo fields in Better Rolls. If you update again, there should still be error messages in the console, but Mob Attack Tool should then ignore ammo and continue on regardless. 

If this patch doesn't work, I'm probably just going to scrap the whole ammo field altogether. I only added it as an afterthought to begin with, after all.

**Comment by Zamrod** [comment]
I imported the Goblin from Roll20. It had ammunition set but didn’t properly refer to the arrows in its inventory (which ammo to use was blank). I set all of them to using no ammunition and it worked. I also set the ammunition properly and it worked.

 Is this feature supposed to only make 1 attack roll even with individual attacks? I got one attack roll and 2 damage rolls. In fact, it seemed to show me an attack roll of 17 but rolled crit damage twice.

Also, it seems to work fine with midi-qol enabled.

**Comment by Stendarpaval** [comment]
> I imported the Goblin from Roll20. It had ammunition set but didn’t properly refer to the arrows in its inventory (which ammo to use was blank). I set all of them to using no ammunition and it worked. I also set the ammunition properly and it worked.

> Also, it seems to work fine with midi-qol enabled.

Good to hear!

> Is this feature supposed to only make 1 attack roll even with individual attacks? I got one attack roll and 2 damage rolls. In fact, it seemed to show me an attack roll of 17 but rolled crit damage twice.

Did the target have an AC of 17? If so, then you probably have turned the `Show Individual Attack Rolls` setting off. When that setting is off, Mob Attack Tool uses an attack roll that matches the target's AC, even if there were one or more critical hits. 

There needs to be at least one attack roll that would hit the target in order for Better Rolls to trigger Midi-QOL's automation features. I should probably remove that attack roll when Mob Attack Tool knows that midi-QOL is turned off, now that I think about it. I'll add that on my todo list. However, the behavior you're seeing is as intended. :)
